lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -153,7 +153,13 @@ def pass_sanity_hook_checks(self, event: EventBase) -> bool:
             return False
 
         if not self.charm.is_relation_feasible(self.relation_name):
-            logger.info("Skipping event %s , relation not feasible.", str(type(event)))
+            if self.charm.status.is_status_related_to_mismatched_revision(
+                self.charm.unit.status.name
+            ):
+                logger.info("Deferring %s. Mismatched versions in the cluster.", str(type(event)))
+                event.defer()
+            else:
+                logger.info("Skipping event %s , relation not feasible.", str(type(event)))
```

Comment:
Nit: Removing nested structure of if statement
```suggestion
      if self.charm.status.is_status_related_to_mismatched_revision(
           self.charm.unit.status.name
      ):
            logger.info("Deferring %s. Mismatched versions in the cluster.", str(type(event)))
            event.defer()
       elif not self.charm.is_relation_feasible(self.relation_name):
           logger.info("Skipping event %s , relation not feasible.", str(type(event)))
```

Reply:
The suggestion doesn't work because the call to `self.charm.status.is_status_related_to_mismatched_revision` requires `self.charm.is_relation_feasible` to [have run](https://github.com/canonical/mongodb-operator/blob/6/edge/src/charm.py#L1621-L1623).
However I agree that this can be improved and I will come up with a not-nested version of this.